### PR TITLE
[SPARK-45192][UI] Fix overdue lineInterpolate parameter for graphviz edge

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

The `edge.lineInterpolate` no longer takes effect for drawing edges. It shall be replaced by d3.curve

### Why are the changes needed?

trivial bugfix for ui.

### Does this PR introduce _any_ user-facing change?

no

#### Before
image
#### After
image

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: core/src/main/resources/org/apache/spark/ui/static/spark-dag-viz.js [hunk 1]
```diff
@@ -328,11 +328,12 @@ function preprocessGraphLayout(g, forJob) {
...
   // Curve the edges
-  var edges = g.edges();
-  for (var j = 0; j < edges.length; j++) {
-    var edge = g.edge(edges[j]);
-    edge.lineInterpolate = "basis";
-  }
+  g.edges().forEach(function (edge) {
+      g.setEdge(edge.v, edge.w, {
+        curve: d3.curveBasis
+      })
```
Comment (yaooqinn) [inline core/src/main/resources/org/apache/spark/ui/static/spark-dag-viz.js:334]:
```suggestion
    g.setEdge(edge.v, edge.w, {
      curve: d3.curveBasis
    })
```

File: core/src/main/resources/org/apache/spark/ui/static/spark-dag-viz.js [hunk 2]
```diff
@@ -328,11 +328,12 @@ function preprocessGraphLayout(g, forJob) {
...
-  for (var j = 0; j < edges.length; j++) {
-    var edge = g.edge(edges[j]);
-    edge.lineInterpolate = "basis";
-  }
+  g.edges().forEach(function (edge) {
+    g.setEdge(edge.v, edge.w, {
+      curve: d3.curveBasis
+    })
+    }
+  )
```
Comment (yaooqinn) [inline core/src/main/resources/org/apache/spark/ui/static/spark-dag-viz.js:336]:
```suggestion
  })
```

File: sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js [hunk 3]
```diff
@@ -109,11 +109,12 @@ function preprocessGraphLayout(g) {
...
-  for (var j = 0; j < edges.length; j++) {
-    var edge = g.edge(edges[j]);
-    edge.lineInterpolate = "basis";
-  }
+  g.edges().forEach(function (edge) {
+      g.setEdge(edge.v, edge.w, {
+        curve: d3.curveBasis
+      })
+    }
+  )
```
Comment (yaooqinn) [inline sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js:117]:
```suggestion
  g.edges().forEach(function (edge) {
    g.setEdge(edge.v, edge.w, {
      curve: d3.curveBasis
    })
  })
```

**Comment by yaooqinn** [comment]
cc @sarutak @HyukjinKwon @dongjoon-hyun thanks

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
I like this!

**Review by sarutak — APPROVED** [review]
+1. Thanks @yaooqinn !

**Pull request closed without merging**